GAE/src/org/akvo/flow/rest/security/oidc/CallbackController.java

```diff
@@ -78,9 +78,11 @@ private void handle(HttpServletRequest req, HttpServletResponse res) throws IOEx
             SecurityContextHolder.getContext().setAuthentication(authentication);
             res.sendRedirect(redirectOnSuccess);
         } catch (AuthenticationException | IdentityVerificationException e) {
-            log.log(Level.SEVERE, e.getMessage());
-            SecurityContextHolder.clearContext();
-            res.sendRedirect(redirectOnFail);
+	    String errorDescriptionId = "error_description";
+	    String errorDescription= req.getParameter(errorDescriptionId);
+	    log.log(Level.INFO, errorDescriptionId+": "+errorDescription+" Error Message: "+e.getMessage());
+	    SecurityContextHolder.clearContext();
+	    res.sendRedirect(redirectOnFail+"?errorCode="+errorDescription);
```

Comment:
Note that this is going to cause the errorCode to be the **String** "null", which the JS code does not handle. Either go back to the if/else statement that was here before, or change the JS code to expect the errorCode null to be "null" string.

Reply:
the js code is already adapted to it

Reply:
I am expecting to see a `if (error == "null")` in the JS side to handle the String "null" that the Java backend is sending.

Is my expectation incorrect?